ts/components/conversation/right-panel/overlay/message-info/OverlayMessageInfo.tsx

```diff
@@ -78,36 +95,133 @@ const StyledMessageDetail = styled.div`
+        });
+        attachmentsWithMediaDetails.push({
+          ...props,
+          duration,
+        });
+      } else if (props?.contentType && isAudio(props.contentType) && !props.duration && props.url) {
+        // eslint-disable-next-line no-await-in-loop
+        const duration = await getAudioDuration({
+          objectUrl: props.url,
+          contentType: props.contentType,
+        });
+
+        attachmentsWithMediaDetails.push({
+          ...props,
+          duration,
+        });
+      } else if (props) {
+        attachmentsWithMediaDetails.push(props);
+      }
+    }
+
+    // This will make the error message for outgoing key errors a bit nicer
+    const errors = (found.get('errors') || []).map((error: any) => {
+      return error;
+    });
+
+    const toRet: MessagePropsDetails = {
+      errors,
+      attachments: attachmentsWithMediaDetails,
+    };
+
+    return toRet;
+  }
+  return null;
+}
+
+function useMessageDetailProps(messageId: string | undefined): MessagePropsDetails | null {
+  const [details, setDetails] = useState<MessagePropsDetails | null>(null);
+
+  const fromState = useMessageAttachments(messageId);
+
+  // this is not ideal, but also doesn't seem to create any performance issue at the moment.
+  // TODO: ideally, we'd want to save the attachment duration anytime we save one to the disk (incoming/outgoing), and just retrieve it from the redux state here.
+  useEffect(() => {
+    let mounted = true;
+    // eslint-disable-next-line more/no-then
+    void getPropsForMessageDetail(messageId, fromState || [])
+      .then(result => {
+        if (mounted) {
+          setDetails(result);
+        }
+      })
+      .catch(window.log.error);
+
+    return () => {
+      mounted = false;
+    };
+  }, [fromState, messageId]);
+
+  return details;
+}
+
 export const OverlayMessageInfo = () => {
+  const dispatch = useDispatch();
+
   const rightOverlayMode = useSelector(getRightOverlayMode);
-  const messageDetailProps = useSelector(getMessageDetailsViewProps);
-  const isDeletable = useMessageIsDeletable(messageDetailProps?.messageId);
+  const messageId = useSelector(getMessageInfoId);
+  const messageDetailProps = useMessageDetailProps(messageId);
+  const isDeletable = useMessageIsDeletable(messageId);
+  const direction = useMessageDirection(messageId);
+  const timestamp = useMessageTimestamp(messageId);
+  const serverTimestamp = useMessageServerTimestamp(messageId);
+  const sender = useMessageSender(messageId);
 
-  const dispatch = useDispatch();
+  // we close the right panel when switching conversation so the convoId of the convoId of that message
+  // is always the currently selected conversation
+  const convoId = useSelectedConversationKey();
 
-  useKey('Escape', () => {
+  const closePanel = useCallback(() => {
     dispatch(closeRightPanel());
     dispatch(resetRightOverlayMode());
-    dispatch(closeMessageDetailsView());
-  });
+    dispatch(closeMessageInfoView());
```

Comment:
Shouldn't resetting the overlay mode clear the message info view already?

Reply:
I looked at the function. Shouldn't we just clear that idea in the `resetRightOverlayMode` function? That way we don't need two functions.

Reply:
updated